ccb/upstream_project.py

```diff
@@ -256,4 +256,54 @@ def _get_owner_repo(cls, recipe):
         raise _Unsupported()
 
 
-_CLASSES = [GithubProject]
+class GnomeProject(UpstreamProject):
+    SOURCE_URL_RE = re.compile(r"https?://((download.gnome.org)|(ftp.gnome.org/pub/gnome))/sources/([^/]+)/")
```

Comment:
You should be able to remove the inner parentheses of your first match group. That will remove the 2nd and 3rd match group and will make the return value of `_get_project` more sensible (currently half of the tuple values are useless)
You can also escape the `.`